From 602b9142a2b7339ff42145065cd829d5d4379aaa Mon Sep 17 00:00:00 2001
From: r1metal <r1metal@localhost>
Date: Wed, 14 Oct 2026 18:21:51 +0200
Subject: [PATCH] wmt: STP, WMT and HIF SDIO lines print at the level they are

The INFO macros of STP, WMT and HIF SDIO print with pr_info, and their
LOUD, DBG, TRC and TRACE macros at KERN_DEBUG, instead of pr_warn. The
WARN macros print with pr_warn.
---
 connectivity/common/common_main/core/include/stp_core.h  | 8 ++++----
 connectivity/common/common_main/include/wmt_exp.h        | 8 ++++----
 connectivity/common/common_main/linux/include/hif_sdio.h | 6 +++---
 3 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/connectivity/common/common_main/core/include/stp_core.h b/connectivity/common/common_main/core/include/stp_core.h
index 2099131..d3fb744 100644
--- a/connectivity/common/common_main/core/include/stp_core.h
+++ b/connectivity/common/common_main/core/include/stp_core.h
@@ -57,12 +57,12 @@ extern INT32 gStpDbgLvl;
 
 #define STP_DBG_FUNC(fmt, arg...)    do {\
 	if (gStpDbgLvl >= STP_LOG_DBG)\
-		osal_warn_print(PFX "%s: " fmt, __func__, ##arg);\
+		printk(KERN_DEBUG PFX "%s: " fmt, __func__, ##arg);\
 } while (0)
 
 #define STP_INFO_FUNC(fmt, arg...)   do {\
 	if (gStpDbgLvl >= STP_LOG_INFO)\
-		osal_warn_print(PFX "%s:[I] " fmt, __func__, ##arg);\
+		pr_info(PFX "%s:[I] " fmt, __func__, ##arg);\
 } while (0)
 
 #define STP_WARN_RATELIMITED_FUNC(fmt, arg...)   do {\
@@ -83,7 +83,7 @@ extern INT32 gStpDbgLvl;
 
 #define STP_TRC_FUNC(f)              do {\
 	if (gStpDbgLvl >= STP_LOG_DBG)\
-		osal_warn_print(PFX "<%s> <%d>\n", __func__, __LINE__);\
+		printk(KERN_DEBUG PFX "<%s> <%d>\n", __func__, __LINE__);\
 } while (0)
 
 #define STP_DUMP_PACKET_HEAD(a, b, c)     do {\
@@ -93,7 +93,7 @@ extern INT32 gStpDbgLvl;
 
 #define STP_TRACE_FUNC(fmt, arg...)       do {\
 	if (gStpDbgLvl >= STP_LOG_DBG)\
-		osal_warn_print(PFX "%s: " fmt, __func__, ##arg);\
+		printk(KERN_DEBUG PFX "%s: " fmt, __func__, ##arg);\
 } while (0)
 
 #define STP_MODE_BIT(x) (0x1UL << x)
diff --git a/connectivity/common/common_main/include/wmt_exp.h b/connectivity/common/common_main/include/wmt_exp.h
index 7941f11..251f2ed 100644
--- a/connectivity/common/common_main/include/wmt_exp.h
+++ b/connectivity/common/common_main/include/wmt_exp.h
@@ -43,12 +43,12 @@
 #define WMT_LOUD_FUNC(fmt, arg...) \
 do { \
 	if (gWmtDbgLvl >= WMT_LOG_LOUD) \
-		osal_warn_print(DFT_TAG "[L]%s:"  fmt, __func__, ##arg); \
+		printk(KERN_DEBUG DFT_TAG "[L]%s:"  fmt, __func__, ##arg); \
 } while (0)
 #define WMT_INFO_FUNC(fmt, arg...)  \
 do { \
 	if (gWmtDbgLvl >= WMT_LOG_INFO) \
-		osal_warn_print(DFT_TAG "[I]%s:"  fmt, __func__, ##arg); \
+		pr_info(DFT_TAG "[I]%s:"  fmt, __func__, ##arg); \
 } while (0)
 #define WMT_WARN_FUNC(fmt, arg...) \
 do { \
@@ -63,12 +63,12 @@ do { \
 #define WMT_DBG_FUNC(fmt, arg...) \
 do { \
 	if (gWmtDbgLvl >= WMT_LOG_DBG) \
-		osal_warn_print(DFT_TAG "[D]%s:"  fmt, __func__, ##arg); \
+		printk(KERN_DEBUG DFT_TAG "[D]%s:"  fmt, __func__, ##arg); \
 } while (0)
 #define WMT_TRC_FUNC(f) \
 do { \
 	if (gWmtDbgLvl >= WMT_LOG_DBG) \
-		osal_warn_print(DFT_TAG "<%s> <%d>\n", __func__, __LINE__); \
+		printk(KERN_DEBUG DFT_TAG "<%s> <%d>\n", __func__, __LINE__); \
 } while (0)
 
 #endif
diff --git a/connectivity/common/common_main/linux/include/hif_sdio.h b/connectivity/common/common_main/linux/include/hif_sdio.h
index bafbc26..474803c 100644
--- a/connectivity/common/common_main/linux/include/hif_sdio.h
+++ b/connectivity/common/common_main/linux/include/hif_sdio.h
@@ -249,15 +249,15 @@ extern INT32 gHifSdioDbgLvl;
 
 #define HIF_SDIO_LOUD_FUNC(fmt, arg...)	\
 do { if (gHifSdioDbgLvl >= HIF_SDIO_LOG_LOUD)	\
-	osal_warn_print(DFT_TAG"[L]%s:"  fmt, __func__, ##arg);	\
+	printk(KERN_DEBUG DFT_TAG"[L]%s:"  fmt, __func__, ##arg);	\
 } while (0)
 #define HIF_SDIO_DBG_FUNC(fmt, arg...)	\
 do { if (gHifSdioDbgLvl >= HIF_SDIO_LOG_DBG)	\
-	osal_warn_print(DFT_TAG"[D]%s:"  fmt, __func__, ##arg);	\
+	printk(KERN_DEBUG DFT_TAG"[D]%s:"  fmt, __func__, ##arg);	\
 } while (0)
 #define HIF_SDIO_INFO_FUNC(fmt, arg...)	\
 do { if (gHifSdioDbgLvl >= HIF_SDIO_LOG_INFO)	\
-	osal_warn_print(DFT_TAG"[I]%s:"  fmt, __func__, ##arg);	\
+	pr_info(DFT_TAG"[I]%s:"  fmt, __func__, ##arg);	\
 } while (0)
 #define HIF_SDIO_WARN_FUNC(fmt, arg...)	\
 do { if (gHifSdioDbgLvl >= HIF_SDIO_LOG_WARN)	\
-- 
2.56.0

